src/main/java/pl/betoncraft/betonquest/objectives/ExperienceObjective.java

```diff
@@ -35,25 +37,39 @@
  */
 public class ExperienceObjective extends Objective implements Listener {
 
-    private final int level;
+    private final int amount;
+    private final boolean level;
 
     public ExperienceObjective(final Instruction instruction) throws InstructionParseException {
         super(instruction);
         template = ObjectiveData.class;
-        level = instruction.getInt();
-        if (level < 1) {
+        this.amount = instruction.getInt();
+        if (amount < 1) {
             throw new InstructionParseException("Level cannot be less than 1");
         }
+        this.level = instruction.hasArgument("level") || instruction.hasArgument("l");
     }
 
     @EventHandler(ignoreCancelled = true)
-    public void onLevelUp(final PlayerLevelChangeEvent event) {
+    public void onLevelChangeEvent(final PlayerLevelChangeEvent event) {
         final String playerID = PlayerConverter.getID(event.getPlayer());
+        onExperienceChange(playerID, event.getNewLevel(), true);
+    }
+
+    @EventHandler(ignoreCancelled = true)
+    public void onExpChangeEvent(final PlayerExpChangeEvent event) {
+        final String playerID = PlayerConverter.getID(event.getPlayer());
+        onExperienceChange(playerID, event.getPlayer().getTotalExperience() + event.getAmount(), false);
+    }
+
+    private void onExperienceChange(final String playerID, final int newAmount, final boolean levelUp) {
         if (!containsPlayer(playerID)) {
             return;
         }
-        if (event.getNewLevel() >= level && checkConditions(playerID)) {
-            completeObjective(playerID);
+        if (level && levelUp || !level && !levelUp){
```

Comment:
Move level check to listeners to keep execution time short and improve readablility?